docs: PE-8622: palette-agent uninstall (#10504)
spectrocloud/librarium: docs/docs-content/deployment-modes/agent-mode/install-agent-host.md

Reviewer: ```suggestion
  export http_proxy=<http-proxy-address>
  export https_proxy=<https-proxy-address>
  export HTTP_PROXY=<http-proxy-address>
  export HTTPS_PROXY=<https-proxy-address>
```

    on your Edge host. Must be enabled. Refer to our
    [Configure networkd to Prepare Host for Overlay Network](./overlay-preparation.md) guide for information on
    installing and configuring `systemd-resolved`.
  - [systemd-networkd](https://www.freedesktop.org/software/systemd/man/latest/systemd-networkd.html) - Required if you
    want Palette to manage static IP addresses or if you plan to use overlay networks for clusters deployed on your Edge
    host. Must be enabled. Refer to our
    [Configure networkd to Prepare Host for Overlay Network](./overlay-preparation.md) guide for information on
    installing and configuring `systemd-networkd`.
  - [conntrack](https://conntrack-tools.netfilter.org/downloads.html) - Required for clusters that use PXK-E as its
    Kubernetes layer.
  - [iptables](https://linux.die.net/man/8/iptables)
  - [rsyslog](https://github.com/rsyslog/rsyslog) - Required for audit logs. Must be enabled.

  If you are using Ubuntu or any OS that uses apt or apt-get for package management, you can issue the following command
  to install all dependencies.

  ```shell
  sudo apt-get update && \

>   sudo apt-get install -y --no-install-recommends \

Reviewer: ```suggestion
  sudo apt-get install --yes --no-install-recommends \
```

     # Cilium: Enable ICMP Type 8 (Echo request) and Type 0 (Echo Reply)
     firewall-cmd --permanent --zone="$ZONE" --add-icmp-block-inversion

     ############### End Cilium Rules ##########################

     # DNS and service communications

     # DNS (CoreDNS)
     firewall-cmd --permanent --zone="$ZONE" --add-port=53/tcp
     firewall-cmd --permanent --zone="$ZONE" --add-port=53/udp

     # Allow inbound/outbound traffic to port 443 (HTTPS)
     firewall-cmd --permanent --zone="$ZONE" --add-port=443/tcp

     # Allow NAT traffic
     firewall-cmd --permanent --add-masquerade

     # Reload firewalld cache
     firewall-cmd --reload
     EOF

     # Make the script executable
     chmod +x firewalld-cilium.sh
     ```

  2. Execute the script with the name of the `firewalld` zone. For example, the following script sets the rules in the
     firewall zone `public`.

     ```shell
     ./firewalld-cilium.sh public
     ```

  </details>

### Enablement

<Tabs groupId="env">

<TabItem value="Central Management Mode">

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Export your Palette registration token.

   ```shell
   export TOKEN=<your-palette-registration-token>
   ```

3. (Proxy only) For Edge hosts that use proxies to access the internet, export the proxy configurations in your current
   terminal session. We recommend exporting the variables both in uppercase and lowercase to ensure compatibility.

>    Replace `<httpProxyAddress>` and `<httpsProxyAddress>` with the address and port to your HTTP and HTTPS proxy

Reviewer: ```suggestion
   Replace `<http-proxy-address>` and `<https-proxy-address>` with the address and port of your HTTP and HTTPS proxy
```

>    export HTTPS_PROXY=<httpsProxyAddress>

Reviewer: ```suggestion
  export http_proxy=<http-proxy-address>
  export https_proxy=<https-proxy-address>
  export HTTP_PROXY=<http-proxy-address>
  export HTTPS_PROXY=<https-proxy-address>
```

:::

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. (Proxy only) For Edge hosts that use proxies to access the internet, export the proxy configurations in your current
   terminal session. We recommend exporting the variables both in uppercase and lowercase to ensure compatibility.

>    Replace `<httpProxyAddress>` and `<httpsProxyAddress>` with the address and port to your HTTP and HTTPS proxy

Reviewer: ```suggestion
   Replace `<http-proxy-address>` and `<https-proxy-address>` with the address and port of your HTTP and HTTPS proxy
```

7. Reboot the host. The host will automatically start the installation process once it reboots.

</TabItem>

</Tabs>

### Validate

<Tabs groupId="env">

<TabItem value="Central Management Mode">

To verify the Palette agent was installed, you can confirm the Edge host is registered with your Palette instance
through the Palette UI or issue `palette-agent` commands directly against the host.

#### Palette UI

1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Clusters**.

3. Select the **Edge Hosts** tab. Locate the machine you installed the Palette agent on, and verify the host is
   **Healthy**.

#### Edge Host Terminal

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Create a symbolic link to the Palette agent binary.

   ```shell
   sudo ln --symbolic /opt/spectrocloud/bin/palette-agent /usr/local/bin/palette-agent
   ```

3. Verify the Palette agent version.

   ```shell
   palette-agent version
   ```

   ```shell title="Example output" hideClipboard
   version: v4.9.a
   build: release
   fips:
   ```

</TabItem>

<TabItem value="Local Management Mode">

To verify the Palette agent was installed, you can either access
[Local UI](../../clusters/edge/local-ui/host-management/access-console.md) using the IP address of your Edge host or
issue `palette-agent` commands directly against the host.

> 

Reviewer: ```suggestion

#### Edge Host Terminal

```

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

    ```

<!-- prettier-ignore-start -->

11. Complete the cluster profile creation process by filling out the remaining layers. Ensure you add the
<VersionedLink text="Registry Connect" url="/integrations/packs/?pack=registry-connect" /> add-on pack. 
This pack is required for clusters in local management mode.

<!-- prettier-ignore-end -->

</TabItem>

</Tabs>

:::warning

If using the FIPS version of Agent Mode on a Rocky Linux Edge host, SELinux may incorrectly label the
`kubeadm-flags.env` file during cluster deployment or when certain configurations are adjusted, preventing the Kubelet
from accessing it and properly managing the cluster. Refer to the
[Edge Troubleshooting Guide](../../troubleshooting/edge/edge.md#scenario---kubelet-process-cannot-access-kubeadm-flags)
for guidance.

:::

### Validate

1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Profiles**.

3. Verify the cluster profile you created is present in the **Cluster Profiles** table.

## Uninstall Palette Agent

The `palette-agent uninstall` command removes the Palette agent and all associated resources from an Edge host. The

> uninstall removes agent binaries, systemd services, Kubernetes components, and runtime state, but some artifacts from

Reviewer: ```suggestion
The `palette-agent uninstall` command removes the Palette agent. The
uninstall process removes agent binaries, systemd services, Kubernetes components, and runtime state, but some artifacts from
```
As it is now, I see a contradiction between "removes the Palette agent and all associated resources" and "some artifacts from the agent or the Kubernetes distribution it provisioned may remain on the filesystem"

| ---------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--force`              | Skip the interactive confirmation prompt. Required for non-interactive or automated use.                                                                                                                                                                                            |
| `--dry-run`            | Display every step that would run without modifying anything on the host.                                                                                                                                                                                                           |

> | `--stylus-root <path>` | Specify the directory where the Palette agent was originally installed. By default, the agent reads this value from `/etc/spectro/environment`. Use this flag if the environment file is missing or corrupt. Most installations use the default location and do not need this flag. |

Reviewer: I'm a bit confused by the phrase "By default, the agent reads this value..."
Is Palette agent meant here?

Author: Yes, thank you! I will specify.

> | `--stylus-root <path>` | Specify the directory where the Palette agent was originally installed. By default, the agent reads this value from `/etc/spectro/environment`. Use this flag if the environment file is missing or corrupt. Most installations use the default location and do not need this flag. |

Author: ```suggestion
| `--stylus-root <path>` | Specify the directory where the Palette agent was originally installed. By default, the Palette agent reads this value from `/etc/spectro/environment`. Use this flag if the environment file is missing or corrupt. Most installations use the default location and do not need this flag. |
```

### Prerequisites

- The [Palette agent installed](#install-palette-agent) on a centrally or locally managed Edge host.

- Root access to the Edge host.

### Enablement

<Tabs groupId="env">

<TabItem value="Central Management Mode">

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Create a symbolic link to the Palette agent binary.

   ```shell
   sudo ln --symbolic /opt/spectrocloud/bin/palette-agent /usr/local/bin/palette-agent
   ```

3. Issue the uninstall command as root. Without flags, the command prompts for confirmation before making changes. Refer

>    to the [table](#uninstall-palette-agent) for additional information on flag usage.

Reviewer: ```suggestion
   to the [table above](#uninstall-palette-agent) for additional information on flag usage.
```

    All 10 tasks completed successfully (3.03s)
   ```

The Edge host record remains in Palette after you uninstall the Palette agent. Even if you reinstall the Palette agent
on the same Edge host, the host cannot be re-registered with Palette until you delete the old record from Palette. This
is because the Palette agent derives its identity from the host's `/etc/machine-id` file, which is an OS-managed file
that the uninstall does not remove. Because the same machine produces the same identity during the reinstall process,
Palette rejects the new registration as a duplicate.

To remove the Edge host from Palette:

1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Clusters**.

3. Select the **Edge Hosts** tab. Locate the machine you uninstalled the Palette agent from. After a few minutes, the
   Edge host becomes **Unhealthy**.

4. Select the three-dot menu beside the Edge host and choose **Delete**. Select **OK** to confirm the deletion.

</TabItem>

<TabItem value="Local Management Mode">

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Create a symbolic link to the Palette agent binary.

   ```shell
   sudo ln --symbolic /opt/spectrocloud/bin/palette-agent /usr/local/bin/palette-agent
   ```

3. Issue the uninstall command as root. Without flags, the command prompts for confirmation before making changes. Refer

>    to the [table](#uninstall-palette-agent) for additional information on flag usage.

Reviewer: ```suggestion
   to the [table above](#uninstall-palette-agent) for additional information on flag usage.
```

   subsequent steps from running.

   On hosts that use a custom install root, all cleanup runs at both the absolute host path and the rooted path under
   `STYLUS_ROOT`.

   ```shell title="Example output" hideClipboard
   time="2026-05-22T18:14:58Z" level=info msg="Creating new cached handler with duration: 30s" version=v4.10.0-260521
   This will permanently remove the Palette Edge Agent and all associated files. Continue? [y/N]: y
    +  Load environment configuration [0ms]
    +  Stop spectro services (/etc/systemd) [2.4s]
    +  Stop spectro services (/run/systemd) [0ms]
    +  Reload systemd daemon [300ms]
    +  Stop bundle mount units [146ms]
    +  Reset node state [0ms]
    +  Remove Kubernetes binaries [0ms]
    +  Remove spectro unit files [0ms]
    +  Remove spectro runtime files [0ms]
    +  Remove stylus root directories [171ms]

    All 10 tasks completed successfully (3.03s)
   ```

</TabItem>

</Tabs>

### Validate

<Tabs groupId="env">

<TabItem value="Central Management Mode">

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Verify the Palette agent has been uninstalled by attempting to check the version.

   ```shell
   palette-agent version
   ```

   A missing Palette agent binary indicates the Palette agent was uninstalled.

   ```shell title="Example output" hideClipboard
   bash: /usr/local/bin/palette-agent: No such file or directory
   ```

</TabItem>

<TabItem value="Local Management Mode">

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

install scripts (central mode) or binaries (local mode), remain on the host; however, certain variables may need to be
re-exported or files extracted, depending on your management mode.

### Prerequisites

<Tabs groupId="env">

<TabItem value="Central Management Mode">

- The required hardware and software dependencies needed to [install the Palette agent](#prerequisites).

- An Edge host that previously had the [Palette agent uninstalled](#uninstall-palette-agent).

</TabItem>

<TabItem value="Local Management Mode">

- The required hardware and software dependencies needed to [install the Palette agent](#prerequisites).

- An Edge host that previously had the [Palette agent uninstalled](#uninstall-palette-agent).

- The Edge host must not be registered at **Clusters** > **Edge Hosts** tab. If it is, select the three-dot menu, and
  **Delete** the Edge host record from Palette.

</TabItem>

</Tabs>

### Enablement

<Tabs groupId="env">

<TabItem value="Central Management Mode">

1.  In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

    ```shell

>     ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

    ```

2.  Export your Palette registration token.

    ```shell
    export TOKEN=<your-palette-registration-token>
    ```

> 3.  (Optional) The `user-data` file remains on the host after uninstalling the Palette agent. View the existing file and

Reviewer: ```suggestion
3.  (Optional) The `user-data` file remains on the host after uninstalling the Palette agent. Review the existing file and
```

    </TabItem>

    </Tabs>

6.  If you downloaded a new Palette agent version, grant execution permissions to the installation script.

    <Tabs groupId="FIPS">

    <TabItem value="Non-FIPS">

    ```shell
    chmod +x ./palette-agent-install.sh
    ```

    </TabItem>

    <TabItem value="FIPS">

    ```shell
    chmod +x ./palette-agent-install-fips.sh
    ```

    </TabItem>

    </Tabs>

7.  Reinstall the agent on your host.

     <Tabs groupId="FIPS">

     <TabItem value="Non-FIPS">

    ```shell
    sudo --preserve-env ./palette-agent-install.sh
    ```

     </TabItem>

     <TabItem value="FIPS">

    ```shell
    sudo --preserve-env ./palette-agent-install-fips.sh
    ```

     </TabItem>

     </Tabs>

</TabItem>

<TabItem value="Local Management Mode">

1. The file `agent-mode-linux-<architecture>.tar` remains on the host after uninstalling the Palette agent. However, you
   must re-extract the agent installation package to the root folder. Replace `<architecture>` with the architecture of
   your CPU. If you have ARM64, use `arm64`. If you have AMD64 or x86_64, use `amd64`.

   ```shell
   sudo tar -xvf agent-mode-linux-<architecture>.tar -C /
   ```

> 2. The file `/var/lib/spectro/userdata` remains on the host after uninstalling the Palette agent. View the existing file

Reviewer: ```suggestion
2. The file `/var/lib/spectro/userdata` remains on the host after uninstalling the Palette agent. Review the existing file
```

   ```shell
   cat /var/lib/spectro/userdata
   ```

3. Run the install command using the staging binary that the tar extraction placed on disk.

   ```shell
   sudo /var/lib/spectro/stylus/opt/spectrocloud/bin/palette-agent install \
       --source dir:/var/lib/spectro/stylus \
       --config /var/lib/spectro/userdata
   ```

4. Reboot the host.

   ```shell
   sudo reboot
   ```

   After the host reboots, the Local UI info screen is displayed and the host is ready for use.

</TabItem>

</Tabs>

### Validate

<Tabs groupId="env">

<TabItem value="Central Management Mode">

To verify the Palette agent was installed, you can confirm the Edge host is registered with your Palette instance
through the Palette UI or issue `palette-agent` commands directly against the host.

#### Palette UI

1. Log in to [Palette](https://console.spectrocloud.com/).

2. From the left main menu, select **Clusters**.

3. Select the **Edge Hosts** tab. Locate the machine you installed the Palette agent on, and verify the host is
   **Healthy**.

#### Edge Host Terminal

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

   ```

2. Create a symbolic link to the Palette agent binary.

   ```shell
   sudo ln --symbolic /opt/spectrocloud/bin/palette-agent /usr/local/bin/palette-agent
   ```

3. Verify the Palette agent version.

   ```shell
   palette-agent version
   ```

   ```shell title="Example output" hideClipboard
   version: v4.9.a
   build: release
   fips:
   ```

</TabItem>

<TabItem value="Local Management Mode">

To verify the Palette agent was installed, you can either access
[Local UI](../../clusters/edge/local-ui/host-management/access-console.md) using the IP address of your Edge host or
issue `palette-agent` commands directly against the host.

1. In your terminal, SSH into the Edge host, substituting the placeholders with the necessary values.

   ```shell

>    ssh -i </path/to/private/key> <username>@<host-ip-or-domain>

Author: no longform

as a cluster node.

<Tabs groupId="env">

<TabItem value="Central Management Mode">

Follow the steps in [Create Cluster Definition](../../clusters/edge/site-deployment/cluster-deployment.md) to deploy a
cluster using your Edge host.

</TabItem>

<TabItem value="Local Management Mode">

Before you can deploy a cluster using your Edge host as a cluster node, you must export your cluster profile as a
cluster definition and, optionally, build and upload a content bundle if your Edge host has restricted internet access.
Follow the below guides in order to deploy your cluster.

1.  [Export Cluster Definition](../../clusters/edge/local-ui/cluster-management/export-cluster-definition.md) - Export a
    cluster definition of your profile. You will use this cluster definition later when you create the cluster in Local
    UI.

2.  [Build Content Bundles](../../clusters/edge/edgeforge-workflow/palette-canvos/build-content-bundle.md) - If your
    host has access to all the images referenced by your cluster profile, you may skip this guide. Otherwise, you must
    build a content bundle that contains all the artifacts required to create your cluster, even if your host has no

>     access to an external image registry.

Reviewer: ```suggestion
    build a content bundle that contains all the artifacts required to create your cluster.
```
Maybe I missed something, but I don't understand the need of this sentence part

Author: No, you're right. It's redundant due to `If your host has access to all the images referenced by your cluster profile`. Removed 👍

3.  [Upload Content Bundles](../../clusters/edge/local-ui/cluster-management/upload-content-bundle.md) - If you built a

>     content bundle, you must upload them to your Edge host.

Reviewer: ```suggestion
    content bundle, you must upload it to your Edge host.
```
